File: src/main/java/localstack/test/config/SqsConfiguration.java
package localstack.test.config;

import com.amazonaws.auth.AWSStaticCredentialsProvider;
import com.amazonaws.auth.BasicAWSCredentials;
import com.amazonaws.client.builder.AwsClientBuilder;
import com.amazonaws.services.sqs.AmazonSQSAsync;
import com.amazonaws.services.sqs.AmazonSQSAsyncClientBuilder;
import org.springframework.cloud.aws.messaging.config.annotation.EnableSqs;
import org.springframework.cloud.aws.messaging.core.QueueMessagingTemplate;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

@Configuration
@EnableSqs
public class SqsConfiguration {

    @Bean
    @Primary
    public AmazonSQSAsync amazonSQSClient() {
        return AmazonSQSAsyncClientBuilder.standard()
                .withEndpointConfiguration(
                        new AwsClientBuilder.EndpointConfiguration("http://localhost:4576", "eu-west-1")
                )
                .withCredentials(
                        new AWSStaticCredentialsProvider(new BasicAWSCredentials("test", "test"))
                )
                .build();
    }

    @Bean
    public QueueMessagingTemplate messagingTemplate(AmazonSQSAsync amazonSqs) {
        return new QueueMessagingTemplate(amazonSqs);
    }
}

File: src/main/kotlin/localstack/test/SqsConfiguration.kt
package localstack.test

import com.amazonaws.services.sqs.AmazonSQS
import org.springframework.beans.factory.BeanFactory
import org.springframework.beans.factory.BeanFactoryAware
import org.springframework.cloud.aws.messaging.config.annotation.EnableSqs
import org.springframework.context.annotation.Configuration
import javax.annotation.PostConstruct

@Configuration
@EnableSqs
open class SqsConfiguration : BeanFactoryAware {

    private var beanFactory: BeanFactory? = null

    override fun setBeanFactory(beanFactory: BeanFactory?) {
        this.beanFactory = beanFactory
    }

    @PostConstruct
    fun init() {
        val bean = beanFactory!!.getBean(AmazonSQS::class.java)
        println(bean)
    }
}